tests/test_checkpoint_io/test_hybrid_parallel_plugin_checkpoint_io.py

```diff
@@ -88,6 +81,9 @@ def _preprocess_data(data):
     with shared_tempdir() as tempdir:
         model_ckpt_path = f"{tempdir}/model"
         optimizer_ckpt_path = f"{tempdir}/optimizer"
+        if not shard:
+            model_ckpt_path = model_ckpt_path + "model.pt"
+            optimizer_ckpt_path = optimizer_ckpt_path + "optimizer.pt"
```

Comment:
What's the purpose of this?

Reply:
when not sharding the checkpoints, the path should be a file, otherwise calling torch.save may cause the program to hang.

Reply:
updated.